### PR TITLE
Breakdowns can support multiple demographics

### PR DESCRIPTION
**Pull request by kkatzen**
#282 
Previously, Breakdowns only supported requests for data broken down by no more than one of age, sex, race.
This PR splits a one-of field into distinct fields.
This PR should not change existing functionality. All VariableProviders should still only support one breakdown for now but can be refactored later to support multiple.

### CODE REVIEW

File: frontend/src/data/Breakdowns.ts [hunk 1]
```diff
@@ -44,23 +44,29 @@ function demographicBreakdownToCol(
 
 export class Breakdowns {
   geography: GeographicBreakdown;
-  // Note: this assumes only one demographic breakdown at a time. If we want to
-  // support more later we can refactor this to multiple boolean fields.
-  demographic?: DemographicBreakdown;
```
Comment (aaronsn) [inline frontend/src/data/Breakdowns.ts:49]: Is the DemographicBreakdown type still used anywhere or can we delete it?

File: frontend/src/data/Breakdowns.ts [hunk 2]
```diff
@@ -149,6 +161,18 @@ export class Breakdowns {
...
+  demographicBreakdownCount() {
+    let demographicBreakdownCount = 0;
+    [this.age, this.sex, this.race, this.race_nonstandard].forEach(
+      (demographic) => {
+        if (demographic) {
+          demographicBreakdownCount++;
+        }
+      }
+    );
+    return demographicBreakdownCount;
```
Comment (aaronsn) [inline frontend/src/data/Breakdowns.ts:173]: you can change this whole block to:
`[this.age, this.sex, this.race, this.race_nonstandard].filter((demo) => !!demo).length`

File: frontend/src/data/variables/BrfssProvider.ts [hunk 3]
```diff
@@ -63,7 +63,8 @@ class BrfssProvider extends VariableProvider {
       !breakdowns.time &&
       (breakdowns.geography === "state" ||
         breakdowns.geography === "national") &&
-      (!breakdowns.demographic || breakdowns.demographic === "race")
+      breakdowns.demographicBreakdownCount() === 1 &&
+      !!breakdowns.race
```
Comment (aaronsn) [inline frontend/src/data/variables/BrfssProvider.ts:67]: Previously this allowed no demographic breakdown - don't you want both `0` and `1` to be valid counts?
```
breakdowns.demographicBreakdownCount() === 0 || (breakdowns.demographicBreakdownCount() === 1 && !!breakdowns.race)
```

File: frontend/src/data/variables/AcsPopulationProvider.ts [hunk 4]
```diff
@@ -39,6 +39,18 @@ class AcsPopulationProvider extends VariableProvider {
...
+      !!metricQuery.breakdowns.race_nonstandard ||
+      !!metricQuery.breakdowns.race
+    ) {
+      return ["acs_population-by_race_state_std"];
+    } else if (!!metricQuery.breakdowns.age) {
+      return ["acs_population-by_age_state"];
+    }
+    return [];
+  }
+
```
Comment (aaronsn) [inline frontend/src/data/variables/AcsPopulationProvider.ts:53]: I think this snuck in from a different PR?

File: frontend/src/data/Breakdowns.ts [hunk 5]
```diff
@@ -44,23 +44,29 @@ function demographicBreakdownToCol(
...
-  // support more later we can refactor this to multiple boolean fields.
-  demographic?: DemographicBreakdown;
   // We may want to extend this to an explicit type to support variants for
   // day/week/month/year.
   time?: boolean;
-
+  race?: boolean;
+  race_nonstandard?: boolean;
+  age?: boolean;
+  sex?: boolean;
```
Comment (aaronsn) [inline frontend/src/data/Breakdowns.ts:53]: What do you think about making these booleans non-optional and then defaulting them to false in the constructor? Avoids the need to add `!!` everywhere they're used

**Review by aaronsn — APPROVED** [review]

File: frontend/src/data/Breakdowns.ts (hunk 1 above)
Comment (kkatzen) [reply frontend/src/data/Breakdowns.ts:49]: It was used in one other place but I don't think it was really needed- just removed!

File: frontend/src/data/Breakdowns.ts (hunk 5 above)
Comment (kkatzen) [reply frontend/src/data/Breakdowns.ts:53]: done- I added a new method on Breakdowns to keep the string in unsupported breakdown error the same - I think it's easier to read when we don't list which are set to false.

**Pull request merged**